### PR TITLE
Reimplement GetHosts with k8s types

### PR DESCRIPTION
**Pull request by honza**
The actual types are coming from metalkube/baremetal-operator.  These
datastructures represent the Kubernetes resources for Day 2.  We are
adding them here for reusability, and compatibility between Day 1 and
Day 2.

### CODE REVIEW

**Comment by honza** [comment]
Here is an example response:

```json
{
  "data": {
    "metadata": {
      "selfLink": "/api/v1/namespace/default/baremetalhosts",
      "resourceVersion": "123456"
    },
    "items": [
      {
        "metadata": {
          "name": "baremetalhost",
          "namespace": "default",
          "uid": "some-uuid",
          "creationTimestamp": "2019-01-22T09:30:00Z",
          "labels": {
            "app": "hello-world"
          }
        },
        "spec": {
          "bmc": {
            "ip": "192.168.100.100",
            "credentialsName": "bmc-creds-valid"
          },
          "online": true
        },
        "status": {
          "lastUpdated": "2019-02-19T19:32:26Z",
          "hardware": {
            "ramGiB": 128,
            "nics": [
              {
                "mac": "00:A0:C9:14:C8:29",
                "ip": "192.168.100.100",
                "speedGbps": 40
              }
            ],
            "storage": [
              {
                "sizeGiB": 1024,
                "info": "disk info"
              }
            ],
            "cpus": [
              {
                "type": "intel",
                "speedGHz": 3
              }
            ]
          },
          "provisioningID": "some ironic ID",
          "image": "image.qcow",
          "goodCredentials": null,
          "errorMessage": ""
        }
      }
    ]
  }
}
```

**Comment by dhellmann** [comment]
We should probably expose a method on the host to return its status, instead of expecting users of the class to look at the label directly.

**Comment by dhellmann** [comment]
> We should probably expose a method on the host to return its status, instead of expecting users of the class to look at the label directly.

See https://github.com/metalkube/baremetal-operator/pull/15

**Comment by honza** [comment]
@dhellmann With the Gopkg.lock change in the operator repo, this PR now imports that code correctly. :100:

File: pkg/server/handler_helpers.go [hunk 1]
```diff
@@ -33,6 +33,15 @@ func RespondWithJson(w http.ResponseWriter, obj interface{}) {
 	}
 }
 
+func RespondWithK8s(w http.ResponseWriter, obj interface{}) {
+	err := json.NewEncoder(w).Encode(obj)
```
Comment (jtomasek) [inline pkg/server/handler_helpers.go]: Can we do this in default RespondWithJson? nesting under `data` seems not necessary.

Comment (honza) [reply pkg/server/handler_helpers.go]: Yes, I agree.

My original thought was that each response would have a "status" and a "data" to make it uniform.

**Comment by honza** [comment]
@jtomasek The build is now failing because of two things:

1. prettier and eslint seem to be in disagreement about the indentation
2. our eslint config isn't compatible with that of openshift/console and won't allow the use of `any`

Do you have any thoughts/ideas on how to do this? Should we relax our rules to be more inline with them? Or should we reimplement the types ourselves more properly/strictly?

**Comment by honza** [comment]
I disabled the troublesome rules in that file only. We should try and keep openshift/console code in separate files so we can ignore these types of errors in isolation.

**Comment by jmolmo** [comment]
Maybe a silly question because probably i'm missing something ... 
Wouldn't it be good to have fields for showing status/errorMessage of each of the different hardware devices? for example, If we have errors from several devices, i only see one place to put this information (status.errorMessage). Maybe to mix all this information here will result confuse for the final user..

**Comment by honza** [comment]
@jmolmo This might be better directed towards the baremetal-operator project where these values are defined. We're simply using it as a dependency.

**Comment by dhellmann** [comment]
@jmolmo it's all still early days. The current data structure is just a placeholder until we know what we really care about. The baremetal operator isn't going to be monitoring devices for errors, though, so we may end up with some other resources to hold more detailed status if we need a place to put errors for individual drives or whatever.

File: src/selectors/hosts.ts [hunk 2]
```diff
@@ -7,7 +7,21 @@ export const getHosts = (state: RootState): Host[] => state.hosts.hosts;
 export const getHostsLoading = (state: RootState): boolean =>
   state.hosts.loading;
 
+const hostToHostTableRow = (host: Host): string[] => {
+  const spec = host.spec ? host.spec : {};
+  const status = host.status ? host.status : {};
```
Comment (jtomasek) [inline src/selectors/hosts.ts]: these 2 lines can be shortened to:   `const { spec = {}, status = {} } = host;`

File: pkg/server/handler_helpers.go (hunk 1 above)
Comment (jtomasek) [reply pkg/server/handler_helpers.go]: Do you want to update RespondWithJson to match this in follow up patch?

File: src/selectors/hosts.ts (hunk 2 above)
Comment (honza) [reply src/selectors/hosts.ts]: Done

File: pkg/integration/hosts.go [hunk 3]
```diff
@@ -15,7 +15,10 @@
 package integration
 
 import (
+	apis "github.com/metalkube/baremetal-operator/pkg/apis/metalkube/v1alpha1"
```
Comment (knowncitizen) [inline pkg/integration/hosts.go]: is this arbitrary?

Comment (honza) [reply pkg/integration/hosts.go]: I'm not sure what you mean.

Comment (knowncitizen) [reply pkg/integration/hosts.go]: Just wondering if it follows a convention.

Comment (honza) [reply pkg/integration/hosts.go]: It most likely does. But this is a link to the actual package. We're only consuming it.

**Pull request merged**